digital-inverter-openlane.ipynb

```diff
@@ -49,58 +49,29 @@
-        "    conda_prefix_path = pathlib.Path('conda-env')\n",
-        "    site_package_path = conda_prefix_path / 'lib/python3.7/site-packages'\n",
-        "    sys.path.insert(0, str(site_package_path.resolve()))\n",
-        "    CONDA_PREFIX = str(conda_prefix_path.resolve())\n",
-        "    PATH = os.environ['PATH']\n",
-        "    LD_LIBRARY_PATH = os.environ.get('LD_LIBRARY_PATH', '')\n",
-        "    %env CONDA_PREFIX={CONDA_PREFIX}\n",
-        "    %env PATH={CONDA_PREFIX}/bin:{PATH}\n",
-        "    %env LD_LIBRARY_PATH={CONDA_PREFIX}/lib:{LD_LIBRARY_PATH}"
-      ]
-    },
-    {
-      "cell_type": "code",
-      "execution_count": null,
-      "metadata": {
-        "id": "ltqhSgAHl4Qt"
-      },
-      "outputs": [],
-      "source": [
-        "%%writefile environment.yml\n",
-        "channels:\n",
-        "  - litex-hub\n",
-        "  - conda-forge\n",
-        "dependencies:\n",
-        "  - open_pdks.sky130a\n",
-        "  - magic\n",
-        "  - openroad\n",
-        "  - netgen\n",
-        "  - yosys\n",
-        "  - gdstk\n",
-        "  - tcllib\n",
-        "  - python\n",
-        "  - pip\n",
-        "  - pip:\n",
-        "    - pyyaml\n",
-        "    - click\n",
-        "    - pandas"
-      ]
-    },
-    {
-      "cell_type": "code",
-      "execution_count": null,
-      "metadata": {
-        "id": "OxONmJT0nP_Y"
-      },
-      "outputs": [],
-      "source": [
-        "!conda env update --prefix $CONDA_PREFIX --file environment.yml"
+        "!curl -Ls https://micro.mamba.pm/api/micromamba/linux-64/latest | tar -xvj bin/micromamba\n",
+        "conda_prefix_path = pathlib.Path('conda-env')\n",
+        "site_package_path = conda_prefix_path / 'lib/python3.7/site-packages'\n",
+        "sys.path.append(str(site_package_path.resolve()))\n",
+        "CONDA_PREFIX = str(conda_prefix_path.resolve())\n",
+        "PATH = os.environ['PATH']\n",
+        "LD_LIBRARY_PATH = os.environ.get('LD_LIBRARY_PATH', '')\n",
+        "%env CONDA_PREFIX={CONDA_PREFIX}\n",
+        "%env PATH={CONDA_PREFIX}/bin:{PATH}\n",
+        "%env LD_LIBRARY_PATH={CONDA_PREFIX}/lib:{LD_LIBRARY_PATH}\n",
+        "!bin/micromamba create --yes --prefix $CONDA_PREFIX\n",
+        "!echo 'python ==3.7*' >> {CONDA_PREFIX}/conda-meta/pinned\n",
+        "!bin/micromamba install --yes --prefix $CONDA_PREFIX \\\n",
+        "                        --channel litex-hub \\\n",
+        "                        --channel main \\\n",
+        "                        open_pdks.sky130a \\\n",
+        "                        magic \\\n",
+        "                        openroad \\\n",
+        "                        netgen \\\n",
+        "                        yosys \n",
+        "!bin/micromamba install --yes --prefix $CONDA_PREFIX \\\n",
+        "                        --channel conda-forge \\\n",
+        "                        tcllib gdstk pyyaml click"
```

Comment:
This gives really nice results - it works way faster than miniconda.

I'm not super familiar with micromamba, but if it does what it says on the tin, then it's a good replacement.